lua/pixelui/core/cl_color.lua

```diff
@@ -130,4 +130,22 @@ function PIXEL.GetRainbowColor()
     lastCol = HSVToColor((time * 50) % 360, 1, 1)
 
     return lastCol
+end
+
+local colorMeta = FindMetaTable("Color")
+
+function colorMeta:Copy()
+    return PIXEL.CopyColor(self)
+end
+
+function colorMeta:Offset(off)
```

Comment:
Surely this should be applying the change to the colour object you're calling this on instead of returning the new one?

Same for the method below.

Reply:
This still applies after the recent changes, these Offset and Lerp should modify the colours they were called with.